### PR TITLE
TextNote Location fixes

### PR DESCRIPTION
**Pull request by moethu**
### Purpose

Fix for TextNotes

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] Snapshot of UI changes, if any.

### Reviewers

@mjkkirschner this is fixing the issue @Benglin discovered.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Libraries/RevitNodes/Elements/TextNote.cs [hunk 1]
```diff
@@ -86,7 +86,14 @@ private void InternalSetType(string value, RVT.TextNoteType type, XYZ location,
         private void InitElement(Autodesk.Revit.DB.TextNote element)
         {
             LocationPoint location = element.Location as LocationPoint;
-            InternalSetType(element.Text, element.TextNoteType, location.Point, element.HorizontalAlignment, location.Rotation.ToDegrees());
+            XYZ position = element.Coord;
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/TextNote.cs]: can you add some comments on what this fix is avoiding?

File: src/Libraries/RevitNodes/Elements/TextNote.cs [hunk 2]
```diff
@@ -135,18 +142,28 @@ private void Init(RVT.View view, RVT.XYZ origin, RVT.HorizontalTextAlignment ali
...
-
-                    if (point.Rotation != rotation.ToRadians())
+                }
+                else
+                {
+                    if (element.Location is LocationPoint)
                     {
-                        point.Rotate(Line.CreateUnbound(XYZ.Zero, XYZ.BasisZ), rotation.ToRadians());
+                        LocationPoint point = (LocationPoint)element.Location;
+
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/TextNote.cs]: same here please add some comments

File: src/Libraries/RevitNodes/Elements/TextNote.cs [hunk 3]
```diff
@@ -135,18 +142,28 @@ private void Init(RVT.View view, RVT.XYZ origin, RVT.HorizontalTextAlignment ali
...
+                        if (!point.Point.Equals(origin))
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/TextNote.cs]: If this is comparing Dynamo Points we should use AlmostEquals

File: src/Libraries/RevitNodes/Elements/TextNote.cs [hunk 4]
```diff
@@ -135,18 +142,28 @@ private void Init(RVT.View view, RVT.XYZ origin, RVT.HorizontalTextAlignment ali
...
+                        {
+                            point.Point = origin;
+                        }
+
+                        if (point.Rotation != rotation.ToRadians())
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/TextNote.cs]: same here - if this is a double comparison it should be done with some epsilon tolerance

**Comment by mjkkirschner** [comment]
@moethu thanks for the quick fix, please add some comments, and can you also add a bit more info to the PR so later when we look back at commit history we know what this is.

**Comment by moethu** [comment]
@mjkkirschner applied the changes

**Comment by Benglin** [comment]
Thanks a ton for fixing this, @moethu! More urgently once this fix is verified, we'll need them in main branches as well as RC branches (I'm so sorry, but there are 8 branches to cross-merge):

https://github.com/DynamoDS/DynamoRevit/tree/Revit2015
https://github.com/DynamoDS/DynamoRevit/tree/Revit2016
https://github.com/DynamoDS/DynamoRevit/tree/Revit2017
https://github.com/DynamoDS/DynamoRevit/tree/Revit2018

https://github.com/DynamoDS/DynamoRevit/tree/RC1.2.0_Revit2015
https://github.com/DynamoDS/DynamoRevit/tree/RC1.2.0_Revit2016
https://github.com/DynamoDS/DynamoRevit/tree/RC1.2.0_Revit2017
https://github.com/DynamoDS/DynamoRevit/tree/RC1.2.0_Revit2018

Thanks again for the quick respond to the issue! 👍

**Comment by mjkkirschner** [comment]
@Benglin these PRs have not been going into the 2015 branches, believe we decided that at the beginning of the project. Just wanted to make sure we're all on same page.

**Comment by Benglin** [comment]
Ah good point, @mjkkirschner, my bad! That will be just 7 branches, since we're still supporting `RC1.2.0_Revit2015` :)

**Comment by mjkkirschner** [comment]
Hey @Benglin - what I mean is that none of @moethu 's PRs are in 2015 (for any of these new Revit nodes) - so I don't think this can be cherry picked there.

**Comment by Benglin** [comment]
Got it, @mjkkirschner. I am going to first merge these into main branches, so @monikaprabhu can test them out right now. Thanks again for reviewing!

**Pull request merged**